coriolis/resources/write_data.c

```diff
@@ -130,10 +133,21 @@ int handle_msg(FILE* input_stream)
     FILE* f = fopen(path, "rb+");
     if (!f)
         return ERR_OPEN_FILE;
+
+    if (fseek(f, 0, SEEK_END))
```

Comment:
What does this seek achieve? Shouldn't seeking 0 bytes from the end of the file always work?
Did you mean to `fseek(f, offset + data_size, SEEK_SET)`?
That should also eliminate the need for the added `if (end_write > disk_size)` at the end. (though it will mean we give out the more generic `ERR_IO_SEEK`)

Reply:
This gives you the disk size. You seek to byte 0 relative to the end, then call ```ftell()``` to give you the offset. That offset is your disk size.

I would rather compare values than try a pythonism in C, where I check for an error rather than compare values.

Reply:
Failed to properly notice the `ftell` below, great point.